test/Tests/src/Data/Text_Spec.enso

```diff
@@ -1116,97 +1124,99 @@ spec =
-            "".locate "" matcher=regex . should_equal (Span.Value (0.up_to 0) "")
-            "".locate_all "" matcher=regex . should_equal [Span.Value (0.up_to 0) ""]
-            "".locate "" matcher=regex mode=Matching_Mode.Last . should_equal (Span.Value (0.up_to 0) "")
+            "".match "foo" matcher=regex . should_equal Nothing
+            "".match "foo" matcher=regex mode=Matching_Mode.Last . should_equal Nothing
+            "".match_all "foo" matcher=regex . should_equal []
+            "".match "" matcher=regex . should_equal ""
+            "".match_all "" matcher=regex . should_equal [""]
+            "".match "" matcher=regex mode=Matching_Mode.Last . should_equal ""
             abc = 'A\u{301}ßC'
-            abc.locate "" matcher=regex . should_equal (Span.Value (0.up_to 0) abc)
-            abc.locate_all "" matcher=regex . should_equal [Span.Value (0.up_to 0) abc, Span.Value (0.up_to 0) abc, Span.Value (1.up_to 1) abc, Span.Value (2.up_to 2) abc, Span.Value (3.up_to 3) abc]
-            abc.locate "" matcher=regex mode=Matching_Mode.Last . should_equal (Span.Value (3.up_to 3) abc)
+            abc.match "" matcher=regex . should_equal abc
+            abc.match_all "" matcher=regex . should_equal ["", "", "", "", ""]
+            abc.match "" matcher=regex mode=Matching_Mode.Last . should_equal ""
 
         Test.specify "should handle overlapping matches as shown in the examples" <|
-            "aaa".locate "aa" mode=Matching_Mode.Last matcher=Text_Matcher.Case_Sensitive . should_equal (Span.Value (1.up_to 3) "aaa")
-            "aaa".locate "aa" mode=Matching_Mode.Last matcher=Regex_Matcher.Value . should_equal (Span.Value (0.up_to 2) "aaa")
+            "aaa".locate "aa" mode=Matching_Mode.Last case_sensitivity=Case_Sensitivity.Sensitive . should_equal (Span.Value (1.up_to 3) "aaa")
 
-            "aaa aaa".locate "aa" mode=Matching_Mode.Last matcher=Text_Matcher.Case_Sensitive . should_equal (Span.Value (5.up_to 7) "aaa aaa")
-            "aaa aaa".locate "aa" mode=Matching_Mode.Last matcher=Regex_Matcher.Value . should_equal (Span.Value (4.up_to 6) "aaa aaa")
+            "aaa aaa".locate "aa" mode=Matching_Mode.Last case_sensitivity=Case_Sensitivity.Sensitive . should_equal (Span.Value (5.up_to 7) "aaa aaa")
 
         Test.specify "should allow to match one or more occurrences of a pattern in the text" <|
             "abacadae".match_all "a[bc]" . should_equal ["ab", "ac"]
             "abacadae".match_all "a." . should_equal ["ab", "ac", "ad", "ae"]
             "abacadae".match_all "a.*" . should_equal ["abacadae"]
             "abacadae".match_all "a.+?" . should_equal ["ab", "ac", "ad", "ae"]
 
+            "aAa".match "aa" mode=Matching_Mode.Last matcher=Text_Matcher.Case_Insensitive . should_equal "Aa"
+            "aAa".match "aa" mode=Matching_Mode.Last matcher=(Regex_Matcher.Value case_sensitivity=Case_Sensitivity.Insensitive) . should_equal "aA"
```

Comment:
@radeusgd I would expect this one to return "Aa"? Is this the difference between Text and Regexp behavor for Matching_Mode.Last that's mentioned in the comments?

Reply:
That's the whole point of this example and the comment.

What is the _actual_ behaviour here? Is the test passing in the current form or not? I'd expect it to be passing in _this_ form, unless something has changed.

```
   ! Last Match in Regex Mode
     Regex always performs the search from the front and matching the last
     occurrence means selecting the last of the matches while still generating
     matches from the beginning. This will lead to slightly different behavior
     for overlapping occurrences of a pattern in Regex mode than in exact text
     matching mode where the matches are searched for from the back.
```

This is explained in the comment that you moved around as part of this. However, maybe the explanation is not that good - if you have ideas how we could rephrase it to make it more understandable, I will appreciate that.

The thing is that Regex does not have a mode for 'matching backwards', all we can do is match all and pick the last one. And also Regex does not return overlapping matched - it will return a match at some position and then continue the search _after_ that match.

I.e. if I do `"abcde".match_all` I will get `["ab", "cd"]` and not for example `["ab", "bc", "cd", "de"]`. So in our case matching `aAa` against `aa` case insensitively has only one match `aA` which is returned.

This is what this test verifies and this is what the doc comment together with that example above tried to explain. Maybe we can improve the explanation to make it more understandable to our users, but I'm not sure how to phrase this stuff better.

Reply:
Yes, that is what I wasn't sure about -- I've added your sentence to the comment: "And also Regex does not return overlapping matched - it will return a match at some position and then continue the search after that match."